cpp/include/copc-lib/io/base_io.hpp

```diff
@@ -14,21 +17,31 @@ class Hierarchy;
 class PageInternal;
 } // namespace Internal
 
+struct CopcConfig;
+
 // Base class for all IO reader/writers
 class BaseIO
 {
   public:
     // Find a node object given a key
     Node FindNode(VoxelKey key);
 
-    // WKT string if defined, else empty
-    std::string GetWkt() const { return this->file_->GetWkt(); }
-    // CopcData
-    las::CopcVlr GetCopcHeader() const { return this->file_->GetCopc(); }
     // Las header
     las::LasHeader GetLasHeader() const { return this->file_->GetLasHeader(); }
+    // CopcInfo
+    CopcInfo GetCopcInfo() const { return this->file_->GetCopcInfo(); }
+    // CopcExtents
+    CopcExtents GetCopcExtents() const { return this->file_->GetCopcExtents(); }
+    // WKT string if defined, else empty
+    std::string GetWkt() const { return this->file_->GetWkt(); }
     // EB Vlr
     las::EbVlr GetExtraByteVlr() const { return this->file_->GetExtraBytes(); }
+    // Config file containing Las Header, Copc Info and Extents, WKT, and EB vlrs
+    CopcConfig GetCopcConfig() const
```

Comment:
duplicating all these is not ideal. my idea for an overarching copc and las config object was so that we didn't have to write all these getters/setters within base_io and copcfile

Reply:
letme know your thoughts on how to approach this